### ISSUE TITLE
RStudio quarto preview "shaking" after knitting quarto page

### ISSUE THREAD

**Issue by shamindras** [rstudio/rstudio#12255]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.07.2
    OS Version      : macOS Monterey 12.2.1
    R Version       : 4.2.1

### Steps to reproduce the problem

Knit only the main [`index.qmd`](https://github.com/shamindras/ss_personal_quarto_blog/blob/main/index.qmd) in and view in RStudio.

### Describe the problem in detail

I'm currently trying to knit and iteratively develop my [quarto website](https://www.shamindras.com/) (source [here](https://github.com/shamindras/ss_personal_quarto_blog)). The **quarto version:** is 1.1.251.

When I knit only the main [`index.qmd`](https://github.com/shamindras/ss_personal_quarto_blog/blob/main/index.qmd) welcome page, RStudio viewer starts "shaking".
I've recorded a `gif` below to illustrate the issue.

![CleanShot 2022-10-29 at 17 24 44](https://user-images.githubusercontent.com/7627188/198860583-2cd8d736-5074-410b-b59f-757731dd154b.gif)

It only does this for this one page, all other `qmd` pages do not shake like this.

I'm using a relatively fast machine with latest R, RStudio, and quarto (see specs below). I've tried restarting the entire computer, but this still persists. I'm not sure what changed to cause this. Did not have the issue until this morning.

Could anyone please help resolve the issue? I'm getting quite dizzy editing the website 😄.

I've flagged this in the [quarto discussion](https://github.com/quarto-dev/quarto-cli/discussions/3104), and was asked to flag it here, see quarto-dev/quarto-cli#3104.

### Describe the behavior you expected

- [X] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by shamindras** [comment]
Just as a follow up, I wanted to mention the point made by @mcanouil [here](https://github.com/quarto-dev/quarto-cli/discussions/3104#discussioncomment-4010045).

> The issue seems to be about the vertical scrolling bar being added or not in the viewer.

So, for some reason on just the main `index.qmd` welcome page, the vertical bar is causing issues in the rstudio viewer pane.

**Comment by ronblum** [comment]
@shamindras Thank you for raising this! In MacOS System Preferences:
1. What do you have set for <kbd>System Preferences -> General -> Show Scroll Bars</kbd>?
2. Does changing this value make a difference?

**Comment by shamindras** [comment]
@ronblum - thanks so much for taking the time to help.

Your suggestion worked! More specifically, I changed my `System Preferences -> General -> Show Scroll Bars` from `Always` to `When scrolling`. That is, I have updated to the following setting:

![CleanShot 2022-11-02 at 13 19 13@2x](https://user-images.githubusercontent.com/7627188/199557957-5239c7af-37fd-4e96-bf32-fb4a9d980ede.png)

The "shaking" effect stopped.

As some feedback for the RStudio IDE, I would prefer to typically show scroll bars `Always`, and it would be nice if the viewer pane could adapt to not shaking, but I'm happy to use the revised settings for now.

**Comment by ronblum** [comment]
Agreed, the viewer pane should not do this. 😄 Now I'll work on reproducing the issue the same steps, but I'm glad that this workaround works!

**Comment by shamindras** [comment]
> Agreed, the viewer pane should not do this. 😄 Now I'll work on reproducing the issue the same steps, but I'm glad that this workaround works!

Thank's @ronblum  - you can try cloning my repo and trying it out that way if it helps 😄 !

**Comment by ronblum** [comment]
@shamindras I haven't been able to reproduce this yet, using your repo; I'll see if it's a difference in the version of MaCOS.

In the meantime, we have a new version of RStudio (Elsbeth Geranium) that we're in the final stages of development and testing, which uses a new framework that resolves a bunch of issues. Would you be willing to try it and see if it resolves this problem? It's located on the dailies page at https://dailies.rstudio.com/rstudio/elsbeth-geranium/electron/macos/. Note, though, that since it's still in development, so it's still meant for testing purposes, not production use.

**Comment by shamindras** [comment]
> In the meantime, we have a new version of RStudio (Elsbeth Geranium) that we're in the final stages of development and testing, which uses a new framework that resolves a bunch of issues. Would you be willing to try it and see if it resolves this problem? It's located on the dailies page at [dailies.rstudio.com/rstudio/elsbeth-geranium/electron/macos](https://dailies.rstudio.com/rstudio/elsbeth-geranium/electron/macos/). Note, though, that since it's still in development, so it's still meant for testing purposes, not production use.

@ronblum Sure thing. In fact I just installed the following RStudio version, per your recommendation:

![CleanShot 2022-11-02 at 17 57 47@2x](https://user-images.githubusercontent.com/7627188/199610811-440b22b6-4bab-43d8-a4fa-c4e84250206e.png)

After rendering my blog in this new RStudio version, the "shaking" stopped despite the Scrolling set to `Always` (my preference). This is good. I also tried restarting the macOS, and this behavior persisted.

In short it _seems_ like everything is working as far as scrolling is concerned. I can keep monitoring and let you know if it arises again.

Thanks for your continued help.

**Comment by ronblum** [comment]
@shamindras I'm glad to hear that this solved the problem—yet one more advantage of switching to Electron. I'm closing this issue as fixed, then. However, if the problem returns, though, please let us know!
